samples/snippets/core/system-text-json/csharp/Utf8ReaderPartialRead.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace SystemTextJsonSamples
+{
+    public class Utf8ReaderPartialRead
+    {
+        public static void Run()
+        {
+            var jsonString = @"{
+                ""Date"": ""2019-08-01T00:00:00-07:00"",
+                ""Temperature"": 25,
+                ""TemperatureRanges"": {
+                    ""Cold"": { ""High"": 20, ""Low"": -10 },
+                    ""Hot"": { ""High"": 60, ""Low"": 20 }
+                },
+                ""Summary"": ""Hot"",
+            }";
+
+            byte[] bytes = Encoding.UTF8.GetBytes(jsonString);
+            var stream = new MemoryStream(bytes);
```

Comment:
In the case of reading from a filestream that might contain a UTF-8 BOM at the start, the caller needs to strip those three bytes from the buffer before passing the bytes to the `Utf8JsonReader`. Otherwise the reader would throw (since the BOM is not considered a valid part of the JSON).

Reply:
Where can I find an example of code that strips the BOM?

Reply:
This isn't exactly applicable here, but it gives the general idea:
https://github.com/dotnet/runtime/blob/b9524694bc8f8b3676a6bcf25eb7586194693b38/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs#L108-L137
https://github.com/dotnet/runtime/blob/b9524694bc8f8b3676a6bcf25eb7586194693b38/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L570-L607

Sorry, I am heads down on some other work, so I can't give a drag-n-drop answer right now. Maybe others on the team can chime in.

A comment might be sufficient for now, to avoid blocking on it.

Reply:
Thanks, I'll go with a comment for now and create an issue to refine the presentation later.